app/javascript/controllers/application.js

```diff
@@ -6,4 +6,31 @@ const application = Application.start()
 application.debug = false
 window.Stimulus   = application
 
+Turbo.setConfirmMethod((message, element) => {
+  let dialog = document.getElementById("turbo-confirm")
+  let msg;
+
+  try {
+    msg = JSON.parse(message)
+  } catch (e) {
+    msg = {
+      title: "Are you sure?",
+      body: "You will not be able to undo this decision",
+      accept: "Confirm"
+    }
+  }
+
+  dialog.querySelector("[id='turbo-confirm-title']").innerHTML = msg.title ? msg.title : "Are you sure?";
+  dialog.querySelector("[id='turbo-confirm-body']").innerHTML = msg.body ? msg.body : "You will not be able to undo this decision";
+  dialog.querySelector("[id='turbo-confirm-accept']").innerHTML = msg.accept ? msg.accept : "Confirm";
```

Comment:
The default messages are defined in the modal template and they are translated (coming from I18n keys). With this logic we are losing the ability to get a default message translated.

I would suggest to conditionally  replace the text if they are provided otherwise keep the default text from the template

Reply:
What about a simple check for the value?  I think the benefit of something like this would be:

- We keep i18n intact
- Developer can "opt-in" to how much customization they need
  - If only title needed, just pass a string: `turbo_confirm: "Are you sure?"`
  - Or... `turbo_confirm: { title: "Are you sure?" }`
  - Or... `turbo_confirm: { title: "...", body: "...", accept: "..." }`

```js
Turbo.setConfirmMethod((message, element) => {
  const dialog = document.getElementById("turbo-confirm");

  try {
    const { title, body, accept } = JSON.parse(message);

    if (title) {
      document.getElementById("turbo-confirm-title").innerHTML = title;
    }

    if (body) {
      document.getElementById("turbo-confirm-body").innerHTML = body;
    }

    if (accept) {
      document.getElementById("turbo-confirm-accept").innerHTML = accept;
    }
  } catch (e) {
    document.getElementById("turbo-confirm-title").innerText = message;
  }

  dialog.showModal();

  return new Promise((resolve, reject) => {
    dialog.addEventListener(
      "close",
      () => {
        resolve(dialog.returnValue == "confirm");
      },
      { once: true }
    );
  });
});
```

Reply:
@adrienpoly @zachgoll Thanks that's pretty much what I was looking to achieve, don't know why I thought overriding the default locales to static strings was a good idea.